client-src/default/utils/createSocketUrl.js

```diff
@@ -72,14 +72,21 @@ function getSocketUrl(urlParts, loc) {
   // all of these sock url params are optionally passed in through
   // resourceQuery, so we need to fall back to the default if
   // they are not provided
-  const host = query.host || hostname;
+  let host = query.host || hostname;
   const path = query.path || '/ws';
   let portOption = query.port || port;
 
   if (portOption === 'location') {
     portOption = loc.port;
   }
 
+  // In case the host is a raw IPv6 address, it can be enclosed in
+  // the brackets as the brackets are needed in the final URL string.
+  // Need to remove those as url.format blindly adds its own set of brackets
+  // if the host string contains colons. That would lead to non-working
+  // double brackets (e.g. [[::]]) host
+  host = host.replace(/^\[(.*)\]$/, '$1');
```

Comment:
Just interesting where we got extra brackets?

Reply:
In the problem case browser is navigated to `http://[::]:8080/` Websocket URL is formed by taking the current `location.hostname` which includes the brackets. Then in the end native `URL.format` method at least on Chrome V8 adds a new set of braces even they already exists.

Feels that this is a V8 issue and gets someday fixed there based on https://github.com/nodejs/node/issues/36654